Aquarius/Src/Aquarius/Core/Layer.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "Aquarius/Core/Utility.h"
+
+#include <deque>
+#include <string>
+
+namespace Aquarius {
+
+	class Event;
+	class Layer;
+
+	using LayerStack = std::deque<uniquePtr<Layer>>;
```

Comment:
Apparently deque doesn't guarantee contiguous storage and is therefore ~50% slower to iterate over than a vector. Although, our bottleneck likely isn't the iteration over the layers but rather actually calling the update method of each layer. Also, we probably won't have enough layers for this to affect us

Reply:
Yeah we can just leave this for now and if it causes issues then switch it to a vector down the road